cli/crates/backend/src/api/schema.rs

```diff
@@ -47,16 +45,35 @@ async fn subgraph_schema(
         .ok_or_else(|| ApiError::SubgraphsError(format!("{response:#?}")))
 }
 
-async fn federated_graph_schema(account: &str, graph: &str, branch: &str) -> Result<Option<String>, ApiError> {
+async fn federated_graph_schema(account: &str, graph: &str, branch: Option<&str>) -> Result<Option<String>, ApiError> {
     let client = create_client().await?;
-    let operation =
-        FetchFederatedGraphSchemaQuery::build(FetchFederatedGraphSchemaArguments { account, graph, branch });
-    let response = client.post(api_url()).run_graphql(operation).await?;
 
-    response
-        .data
-        .as_ref()
-        .and_then(|data| data.branch.as_ref())
-        .ok_or_else(|| ApiError::SubgraphsError(format!("{response:#?}")))
-        .map(|branch| branch.schema.clone())
+    if let Some(branch) = branch {
+        let operation =
+            FetchFederatedGraphSchemaQuery::build(FetchFederatedGraphSchemaArguments { account, graph, branch });
```

Comment:
I may be missing some reason why it doesn't work, but seems like you could have made `FetchFederatedGraphSchemaArguments` take a `branch: Option<&str>` and reused it for both of these cases.

Not a big deal though

Reply:
I'm not sure what it would look like, since the branch is required on `Query.branch`. We would need to branch on whether the branch is provided or not. Maybe there's another field in the API that would work?

Reply:
Ah yeah - I was indeed missing a reason why it wouldn't work.  Ignore this comment